libs/platform/src/lib/shared/domain/data-source.ts

```diff
@@ -174,18 +185,36 @@ export class ComboBoxDataSource<T> implements DataSource<T> {
             searchParam.set('limit', ComboBoxDataSource.MaxLimit);
         }
 
-        this.dataProvider.fetch(searchParam).subscribe((result: T[]) => {
-            this.dataChanges.next(result);
-        });
+        this.dataProvider
+            .fetch(searchParam)
+            .pipe(takeUntil(this._onDestroy$))
+            .subscribe(
+                (result: T[]) => {
+                    this._onDataReceived$.next();
+                    this._dataLoading = false;
+                    this._dataChanges.next(result);
+                },
+                () => {
+                    this._onDataReceived$.next();
+                    this._dataLoading = false;
```

Comment:
Shouldn't we add `this._dataChanges.error(error);`?

Reply:
There's no error handling in any of the components and I'm not sure it should even be there, because any error popups and similar things should be handled outside of these components